internetarchive/openlibrary
# Implemented Richer UI on App Download from Chrome

PR Template
What issue does this PR close?

Closes #8929

Worked with @seckhen and @NickDevi on this issue.

What does this PR achieve?

Feature: Adds richer install UI for Progressive Web App (PWA) by incorporating screenshots for both desktop and mobile, along with a description.

Technical
This PR updates the manifest file to include screenshots for different devices (desktop and mobile). It utilizes the form_factor property to differentiate between wide (desktop) and narrow (mobile) screens, ensuring compatibility with Chrome's richer install UI.

Screenshots were added for the following contexts:

Desktop views of key pages like Main Page, Search Books, and Student Library.
Mobile views of the same pages optimized for narrow displays.
The manifest.json file was updated, and screenshots were uploaded to static folder.

Testing
To verify:

Clone the repository and switch to this branch.
Link the updated manifest.json to your development build.
Use Chrome to inspect the richer install UI for PWAs:
Open chrome://inspect/#apps.
Check that all screenshots are visible for both desktop and mobile contexts.
Test the UI responsiveness by viewing the install banner on different screen sizes.
Screenshot
Evidence of functionality:

Before:
<img width="1470" alt="Screenshot 2024-12-12 at 3 57 20 AM" src="https://github.com/user-attachments/assets/a349bb2c-7cce-4296-9bf1-1aea5c2efb31" />

After Desktop screenshots: 
<img width="1470" alt="image" src="https://github.com/user-attachments/assets/b35728c5-9331-4296-8999-f032441c81b0" />

After Mobile screenshots: 
<img width="494" alt="image" src="https://github.com/user-attachments/assets/eb0493e8-7595-4f88-8547-d43b18484216" />

Stakeholders
@RayBB (Lead)
